### PR TITLE
feat: POI creation screen improvements

### LINKED ISSUE #1870: POI creation screen needs to be ported to the grid layout
Otherwise it will break like the old party management screen on larger gui scales

### PR DESCRIPTION
**Pull request by ryanzhoudev**
- Moves it to the grid system to work on all resolutions and gui scales (Closes #1870)
- map preview (see video)
- Fixes map being dragged when mouse is not actually touching the map

https://github.com/Wynntils/Artemis/assets/57310593/24d7f958-502c-48af-971c-e7636c96c54c

### CODE REVIEW

**Comment by kristofbolyai** [comment]
You might need to render a bit wider, the right edge has a black line
image

File: common/src/main/java/com/wynntils/screens/maps/PoiCreationScreen.java [hunk 1]
```diff
@@ -282,70 +326,115 @@ protected void doInit() {
...
+                    McUtils.mc().renderBuffers().bufferSource();
+            poi.renderAt(
+                    poseStack,
+                    bufferSource,
+                    MapRenderer.getRenderX(poi, mapCenterX, centerX, currentZoom),
+                    MapRenderer.getRenderZ(poi, mapCenterZ, centerZ, currentZoom),
+                    hovered == poi,
+                    1,
+                    currentZoom);
+            bufferSource.endBatch();
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/screens/maps/PoiCreationScreen.java]:
```suggestion
            Poi poi = new CustomPoi(
                    new PoiLocation(parsedXInput, null, parsedZInput),
                    nameInput.getTextBoxInput(),
                    CustomColor.fromHexString(colorInput.getTextBoxInput()) == CustomColor.NONE
                            ? CommonColors.WHITE
                            : CustomColor.fromHexString(colorInput.getTextBoxInput()),
                    Services.Poi.POI_ICONS.get(selectedIconIndex),
                    selectedVisiblity);
                    
            MultiBufferSource.BufferSource bufferSource =
                    McUtils.mc().renderBuffers().bufferSource();
                   
            poi.renderAt(
                    poseStack,
                    bufferSource,
                    MapRenderer.getRenderX(poi, mapCenterX, centerX, currentZoom),
                    MapRenderer.getRenderZ(poi, mapCenterZ, centerZ, currentZoom),
                    hovered == poi,
                    1,
                    currentZoom);
                    
            bufferSource.endBatch();
```

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/maps/PoiCreationScreen.java]: (bit of spacing)

**Comment by kristofbolyai** [comment]
The code looks good. Do you think we should add some kind of interaction with the map? Like shift-right click to set the coordinates..?

**Comment by kristofbolyai** [comment]
Or if you opt to merge it as is, it is fine. Let me know.

**Comment by ryanzhoudev** [comment]
> The code looks good. Do you think we should add some kind of interaction with the map? Like shift-right click to set the coordinates..?

I did it with middle click to more closely match the main map. We should actually consider syncing all of these to one control schema; I remember the guild map being very different compared to the main map for mouse buttons. Not sure about seaskipper.

Also I am ok with merging now.

**Review by kristofbolyai — APPROVED** [review]

**Comment by magicus** [comment]
I think we should skip all these "Press shift+alt and double click middle button" stuff, and just do "press right mouse button for a popup context menu". It will require a bit more work to get a context menu framework in place, but it will be much more user friendly.

**Pull request merged**